python/rascaline/rascaline/utils/splines.py

```diff
@@ -0,0 +1,396 @@
+"""
+Splined radial integrals
+========================
+
+Classes for generating spline points which can be used as input for the for the radial
+integral hyper paramater option. For an complete example for the LE how to use the
+classes see :ref:`example-splines`.
+
+.. autoclass:: rascaline.utils.SplinerBase
+    :members:
+    :show-inheritance:
+
+.. autoclass:: rascaline.utils.SplinerFromFunction
+    :members:
+    :show-inheritance:
+"""
+
+from abc import ABC, abstractmethod
+from typing import Callable, Dict, Optional, Union
+
+import numpy as np
+
+
+class SplinerBase(ABC):
```

Comment:
since this is specifically for tabulated radial integral, I think it should be reflected somewhere in the name:
```suggestion
class RadialIntegralSplinerBase(ABC):
```